docs/src/main/sphinx/glossary.rst

```diff
@@ -0,0 +1,187 @@
+
+Cluster
+    A Trino cluster provides the resources to run queries against numerous data
+    sources. Clusters define the number of workers, the configuration for the
+    JVM runtime, configured data sources, and others aspects. For more
+    information, see :ref:`trino-concept-cluster`.
+
+.. _glossConnector:
+
+Connector
+    Translates data from a data source into Trino schemas, tables, columns,
+    rows, and data types. A :doc:`connector </connector>` is specific to a data
+    source, and is used in catalog <glossCatalog>_ configurations to define what
+    data source the catalog connects to.
+
+Container
+    A lightweight virtual package of software that contains libraries, binaries,
+    code, configuration files, and other dependencies needed to deploy an
+    application. A running container does not include an operating system,
+    instead using the operating system of the host machine. To learn more, read
+    read about `containers <https://kubernetes.io/docs/concepts/containers/>`_
+    in the Kubernetes documentation.
+
+.. _glossDataVirtualization:
+
+Data virtualization
+    `Data virtualization <https://en.wikipedia.org/wiki/Data_virtualization>`_
+    is a method of abstracting an interaction with multiple
+    :ref:`heterogeneous data sources <glossDataSource>`, without needing to know
+    the distributed nature of the data, its format, or any other technical
+    details involved in presenting the data.
+
+.. _glossDataSource:
+
+Data source
+    A data source is a system, such as PostgreSQL or Iceberg on S3 data, from
+    which data is retrieved. In Trino, users query data sources with
+    `catalogs <glossCatalog>`_ that connect to each source. See
+    :ref:`trino-concept-data-sources` for more information.
+
+.. _glossGzip:
+
+gzip
+    `gzip <https://en.wikipedia.org/wiki/Gzip>`_ is a compression format and
+    software that compresses and decompresses files. This format is used several
+    ways in Trino, including deployment and compressing files in
+    :ref:`object storage <glossObjectStorage>`. The most common extension for
+    gzip-compressed files is ``.gz``.
+
+.. _glossHDFS:
+
+HDFS
+    `Hadoop Distributed Filesystem (HDFS) <https://en.wikipedia.org/wiki/Apache_Hadoop#HDFS>`_
+    is a scalable :ref:`open source <glossOpenSource>` filesystem that was one
+    of the earliest distributed big data systems created to store large amounts
+    of data for the
+    `Hadoop ecosystem <https://en.wikipedia.org/wiki/Apache_Hadoop>`_.
+
+.. _glossJKS:
+
+Java KeyStore (JKS)
+    The system of public key cryptography supported as one part of the Java
+    security APIs. The legacy JKS system recognizes keys and certificates stored
+    in *keystore* files, typically with the ``.jks`` extension, and relies on a
+    system-level list of CAs in *truststore* files installed as part of the
+    current Java installation.
+
+Key
+    A cryptographic key specified as a pair of public and private keys.
```

Comment:
Realizing now that this definition is "a key is a pair of keys" and maybe that should be improved. Would "..public and private key strings" be accurate?

Reply:
let's just say public and private strings used.